python/tests/test_types.py

```diff
@@ -161,3 +191,46 @@ def test_nested_field():
     assert field_var.field_id == 1
     assert isinstance(field_var.type, StructType)
     assert str(field_var) == str(eval(repr(field_var)))
+
+
+@pytest.mark.parametrize(
+    "input_index,input_type",
+    [
+        (1, BooleanType),
+        (2, IntegerType),
+        (3, LongType),
+        (4, FloatType),
+        (5, DoubleType),
+        (6, DateType),
+        (7, TimeType),
+        (8, TimestampType),
+        (9, TimestamptzType),
+        (10, StringType),
+        (11, UUIDType),
+        (12, BinaryType),
+    ],
```

Comment:
can you use iteratools.permutations and one list to avoid the duplication?

Reply:
Let me check it.

Reply:
@emkornfield  Instead of using `iteratools`, I defined a non_parameterized_types variable and then removed all duplications in this test and also another test as well.

Reply:
Looks great.  Thanks!